src/components/GifTile.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react'
+import { NavLink } from 'react-router-dom'
+import styled from 'styled-components';
+import { connect } from 'react-redux'
+
+class GifTile extends Component {
+  constructor(props) {
+    super(props);
+    this.showControls = this.showControls.bind(this);
+    this.hideControls = this.hideControls.bind(this);
+    this.save = this.save.bind(this);
+    this.remove = this.remove.bind(this);
+    this.state = {
+      showControls: false
+    };
+  }
+
+  render () {
+    return (
+      <Tile onMouseOver={this.showControls} onMouseLeave={this.hideControls}>
+        {
+          this.props.isFavorite(this.props.data.id) &&
+          <IconFavorite>
+            <span role="img" aria-label="favorite">⭐</span>
+          </IconFavorite>
+        }
+        <NavLink to={'/details/' + this.props.data.id}>
```

Comment:
The only extra thing about `NavLink` vs `Link` is that NavLink adds extra attributes when it matches the route, In this case we could use only `Link` to redirect